orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -105,172 +105,120 @@ public long getTimeout() {
   @Nonnull
   @Override
   public TaskResult execute(@Nonnull StageExecution stage) {
-    Map<String, String> rollbackTypes = new HashMap<>();
-    Map<String, Map> rollbackContexts = new HashMap<>();
-
     StageData stageData = stage.mapTo(StageData.class);
-    Map<String, Object> cluster;
-
     AtomicReference<Moniker> moniker = new AtomicReference<>(stageData.moniker);
-    if (moniker.get() == null && stageData.serverGroup != null) {
-      try {
-        Map<String, Object> serverGroup =
-            retrySupport.retry(
-                () ->
-                    fetchServerGroup(
-                        stageData.credentials, stageData.regions.get(0), stageData.serverGroup),
-                5,
-                1000,
-                false);
 
-        moniker.set(objectMapper.convertValue(serverGroup.get("moniker"), Moniker.class));
-      } catch (Exception e) {
-        logger.warn(
-            "Failed to fetch server group, retrying! (account: {}, region: {}, serverGroup: {})",
-            stageData.credentials,
-            stageData.regions.get(0),
-            stageData.serverGroup,
-            e);
-        return TaskResult.RUNNING;
-      }
+    boolean success =
```

Comment:
To use a consistent approach to communicate failed returns, maybe return the moniker or `null` here instead of a boolean?

Reply:
Yeah, that was my initial implementation, but the downside there is that it's less clear that the monitor object being passed in is being returned.

I also considered throwing an exception (which would be the more Java-ish way), but I didn't think that would improve readability.

Reply:
To clarify, I meant you would instantiate the moniker object inside the function instead of populating a pre-created one, since there's nothing done to that object after creation right above, in other words, there's no advantage to creating it outside the function.

Reply:
Ohhhh! Yes, that's definitely better. I'll do that.

Reply:
Fuxed un 0d664470d